src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -41,6 +42,56 @@ public AsmMethodRemapper(MethodVisitor mv, Remapper remapper) {
 			super(mv, remapper);
 		}
 
+		private String baseTypeName(String descriptor) {
+			switch (descriptor.charAt(0)) {
+				case 'B':
+					return "byte";
+				case 'C':
+					return "char";
+				case 'D':
+					return "double";
+				case 'F':
+					return "float";
+				case 'I':
+					return "int";
+				case 'J':
+					return "long";
+				case 'S':
+					return "short";
+				case 'Z':
+					return "boolean";
+				case 'V':
+					return "void";
+				case '[':
+					int idx = 1;
+					while(descriptor.charAt(idx) == '[') idx += 1;
+					String s = "array";
+					if(idx > 1) s += idx + "D";
+
+					return s + titleCase(baseTypeName(descriptor.substring(1)));
+				case 'L':
+					String name = descriptor.substring(1, descriptor.length() - 1);
+					name = remapper.mapType(name);
+					return name.substring(name.lastIndexOf('/')+1, name.length());
+
+			}
+			return "var";
+		}
+
+		private String titleCase(String s) {
+			char[] chars = s.toCharArray();
+			chars[0] = Character.toTitleCase(chars[0]);
+			return new String(chars);
+		}
+
+		@Override
+		public void visitLocalVariable(String name, String descriptor, String signature, Label start, Label end, int index) {
+			if(name.startsWith("\u2603")) {
+				name = baseTypeName(descriptor) + "_" + index;
+			}
+			super.visitLocalVariable(name, descriptor, signature, start, end, index);
```

Comment:
I'd copy the code from MethodRemapper.visitLocalVariable and MethodVisitor.visitLocalVariable here without any remaining super calls. This way the name generation can use the final descriptor and we get closer to what needs to be done for mapping based LV renaming anyway.

Reply:
Right, this is how I found out about `Type`. It's calling `this.mv.visitLocalVariable` now instead of `super.visitLocalVariable`